console/src/main/java/org/georchestra/console/ws/passwordrecovery/NewPasswordFormController.java

```diff
@@ -106,9 +101,9 @@ public String setupForm(@RequestParam("token") String token, Model model) throws
             return "newPasswordForm";
 
         } catch (NameNotFoundException e) {
-
-            return "passwordRecoveryForm";
-
+            model.asMap().clear();
+            request.getSession().setAttribute("errmsg", "bad.token");
+            return "redirect:/account/passwordRecovery";
```

Comment:
As only session is used, we could use HttpSession instead.
`public String setupForm(@RequestParam(name = "token", required = false) String token, Model model,
            HttpSession session) throws IOException `
 and 
 `session.setAttribute("errmsg", "bad.token")`
 but it's definitely not a mandatory thing.

Reply:
you mean, passing the session object directly instead of the whole request ?

Reply:
after having discussed, it is ok to leave it as it is